packages/components/src/navigator/router/with-router.js

```diff
@@ -0,0 +1,40 @@
+/**
+ * External dependencies
+ */
+import hoistNonReactStatics from 'hoist-non-react-statics';
+
+/**
+ * Internal dependencies
+ */
+import RouterContext from './router-context';
+
+// A public higher-order component to access the imperative API
+function withRouter( Component ) {
```

Comment:
Do we need this? Why can't we just use `useRouter`?

Reply:
I don't think we'll need it, but it was added likely due to the fact there are a bunch of `Component` instances lying around.